sdk/Sdk.Generators/FunctionMetadataProviderGenerator/FunctionMetadataProviderGenerator.Emitter.cs

```diff
@@ -110,6 +119,52 @@ private string AddBindingInfo(string bindingListVariableName, IList<IDictionary<
 
                 return builder.ToString();
             }
+
+            private bool TryBuildRetryOptions(GeneratorRetryOptions? retry, out string? retryOptionsAsString)
```

Comment:
Consider passing in the existing `StringBuilder` and appending directly. Would be a good idea to follow that practice everywhere. Also don't need a `bool` return, because if it already has the `StringBuilder` it can just return without appending anything.

Generators can be run in a tight loop in your editor, so bad perf will impact developer experience. The more objects you allocate, the more GC needs to run, which impacts perf.